mutual_debt/main.py

```diff
@@ -0,0 +1,51 @@
+"""Mutual Debt Simplification
+
+Usage:
+  simplify-debts <data_file>
+  simplify-debts (-h | --help)
+
+Options:
+  -h --help     Show this screen.
+"""
+import json
+import sys
+
+from mutual_debt.mutual_debt_simplification import debt_list_to_graph, \
+    simplify_debt_graph, draw_graph
+
+
+def print_error(*args, sep=' ', end='\n'):
+    """ Prints values to the stderr stream """
+    print("ERROR:", *args, sep, end, file=sys.stderr)
+
+
+def main():
+
+    if len(sys.argv) != 2:
+        print(__doc__)
+        sys.exit(1)
+
+    if sys.argv[1] == '-h' or sys.argv[1] == '--help':
+        print(__doc__)
+        return
+
+    data_file = sys.argv[1]
+
+    # Try to load debts from data file
+    # On failure: show error and exit using error code
+    try:
+        with open(data_file) as file:
+            debts = json.load(file)
+
+    except IOError as error:
+        print_error("failed to read data file:", str(error))
+        sys.exit(1)
+
+    initial_debt_graph = debt_list_to_graph(debts['debt_list'], debts['names'])
+    draw_graph(initial_debt_graph, 'Initial_Mutual_Debt', open_file=False)
+    simplified_debt_graph = simplify_debt_graph(initial_debt_graph)
+    draw_graph(simplified_debt_graph, 'Simplified_Mutual_Debt')
+
+
+if __name__ == '__main__':
```

Comment:
There's no reason to import this file, so why define main() instead of just running the code?

Reply:
I did not understand what you meant "import this file". Are you talking about main.py? If so, then where is this file imported?

There are three reasons for using the main() function:
  1. it is standard way to do it in python;
  2. it avoids problems with shadowing names;
  3. and most important, to define a console script you have to point to a function. If you look in the setup.py file, you'll see that the console script 'simplify-debts' takes a path to a function, the main() function. As far as I know this is the only way to do it.

Reply:
I meant that this isn't imported, so there was no need for the `if __name__ ==  '__main__'` technique. 
I didn't know about point 3, though, so nevermind :)

Reply:
Yes, you're right: main.py is not meant to be used as a module and should not be imported. The use of that technique is redundant in this case. Nevertheless, I believe it is still good practice to use it.